### ISSUE TITLE
Software Updates cannot be disabled

### ISSUE THREAD

**Issue by Morasithil** [safing/portmaster#1329]
**Pre-Submit Checklist**:

- Check applicable sources for existing issues:
  - [Windows Known Issues](https://docs.safing.io/portmaster/install/windows#known-issues)
  - [Linux Known Issues](https://docs.safing.io/portmaster/install/linux#compatibility)
  - [Github Issues](https://github.com/safing/portmaster/issues?q=is%3Aissue+label%3Abug)

**What happened**:
A friend of mine uses a pc with portmaster on it, i installed portmaster on his system because he doesn't know much about computers but he cares about his privacy. I've installed portmaster in combination with simplewall.
The problem is... i've disabled automatic software updates in portmaster but it updated itself anyway and this happened atleast three times as of now, the issue is when using portmaster combined with simplewall, when portmaster updates itself the new portmaster-version.exe must first be whitelisted in simplewall. If this step isn't taken then portmaster won't work correctly and some websites fail to load. I had to manualy whitelist the new portmaster exe on my friends pc to get it working again.

TLDR: I hope in the future when selecting "automatic software updates - disabled" in portmaster that this will be enforced properly and not straight up ignored. (windows 10 21H2)

**What did you expect to happen?**:

**How did you reproduce it?**:

**Debug Information**:

**Comment by Raphty** [comment]
which one did you select?
![image](https://github.com/safing/portmaster/assets/11504285/6cf448cb-ab99-4f52-9067-364b4cf257e6)

**Comment by Morasithil** [comment]
i use only automatic intelligence updates however the portmaster updated to a full new version even with the setting disabled, my impression was a missconfiguration. i'll add the debug later

**Comment by Raphty** [comment]
From what version to which one did that happen?

**Comment by dhaavi** [comment]
Hey @Morasithil, I have found the issue: Disabling software updates in itself works correctly, but the version check that Portmaster uses forced a download anyway. So it worked for all assets _except_ for the core.

The referenced PR fixes this and will go into the beta release channel today or tomorrow.
